packages/services/schema/src/lib/compose.ts

```diff
@@ -315,57 +346,74 @@ async function callExternalService(
   });
 
   try {
-    logger?.debug('Calling external composition service (url=%s)', input.url);
+    logger?.debug(
+      'Calling external composition service (url=%s, timeout=%s)',
+      input.url,
+      timeoutMs,
+    );
+
     const response = await got(input.url, {
       method: 'POST',
       headers: input.headers,
       body: input.body,
       responseType: 'text',
       retry: {
-        limit: 5,
+        limit: 3,
         methods: ['POST', ...(got.defaults.options.retry.methods ?? [])],
         statusCodes: [404].concat(got.defaults.options.retry.statusCodes ?? []),
         backoffLimit: 500,
       },
       timeout: {
         request: timeoutMs,
-        // connecting should be quick
-        lookup: 10_000,
-        connect: 10_000,
-        secureConnect: 10_000,
       },
     });
 
     span.setAttribute('http.response.body.size', response.rawBody.length);
     span.setAttribute('http.response.status_code', response.statusCode);
     span.end();
 
-    return JSON.parse(response.body) as unknown;
+    return { type: 'success', data: JSON.parse(response.body) as unknown };
   } catch (error) {
     if (error instanceof RequestError) {
       if (!error.response) {
-        span.setAttribute('error.message', error.message);
+        let message = error.message;
+
+        if (!message && error.cause instanceof AggregateError) {
+          message = error.cause.message;
```

Comment:
Should we collect the `error.errors` messages instead? For an aggregate error, there may be multiple messages that would be helpful.

Reply:
Usually the error messages are the same for each attempt. I already have some notes to further improve the whole flow. We can do this in a follow up pr.